### DISCUSSION TITLE
Match file name

### DISCUSSION THREAD

**Discussion by rodhash** [kyazdani42/nvim-web-devicons#493] [Q&A]
This plugin support something like this?

```lua
        ["webpack.*.js"] = {
```

Found nothing similar in other issues / discussion / docs so raising this issue. The idea is to set a particular icon if the filename starts with "something" or in this example, "webpack".

**Comment by gegoune** [comment]
It's in readme: `override_by_filename`.

**Reply by rodhash** [reply]
It doesn't seem to work. Here is my test:

```lua
        ["foo*.baz"] = {
          icon = "",
          color = "#46D146",
          name = "foobar"
        },
```

**Reply by gegoune** [reply]
I am not sure from top of my head if it takes wildcards. Try full name first to see if it works.

**Comment by rodhash** [comment]
Can you pls point where exactly it's covered in the docs this kind of usage? I'm not finding.

I've found documentation for using the full filename but not "foo*bar" or something like this.

**Comment by Yash-xoxo** [comment]
- Builds on the maintainer's response - Acknowledges @alex-courtis's clarification and expands on it with practical examples
- Directly solves the user's problem - Provides working code specifically for TelescopePrompt buffers
- Explains the root cause - Clarifies why their original approach with set_icon_by_filetype didn't work (it expects existing icon names, not new definitions)
- Offers multiple solutions - Shows the two-step process, a complete working example, and an all-in-one alternative
- Includes debugging help - Provides commands to verify filetype and check if icons are registered correctly
- Explains the API conceptually - Breaks down the difference between definition functions and reference/mapping functions
- Addresses their specific example - Shows the corrected version for their .norg file test case
